BANDWITH: Add DYNAMIC option to be able to autoselect default interface

erebe opened the pull request:
I often switch between Ethernet and wifi, so it is always a pain to manually change the config file when It happens.

I added an special interface name "DYNAMIC" that when is use will force yabar to take the interface of the default route of your computer.
It fallback to loopback interface if none is found.

erebe commented:
Any interest in it ?

jacksgt commented on src/intern_blks/ya_intern.c:
```diff
@@ -207,62 +207,74 @@ void ya_int_brightness(ya_block_t *blk) {
+        }
+        if_name[len-1] = '\0';
+
+fail:
```
Please remove the goto, that's kind of overkill here :-)

erebe replied:
k

jacksgt commented on src/intern_blks/ya_intern.c:
```diff
@@ -207,62 +207,74 @@ void ya_int_brightness(ya_block_t *blk) {
-		if(fscanf(txfile, "%lu", &tx)!=1)
-			fprintf(stderr, "Error getting values from file (%s)\n", txpath);
+	for(;;) {
+                if(strcmp(blk->internal->option[0], "DYNAMIC") == 0) {
```
Ideally I'd like to make this the default, i.e. rather than having to type `DYNAMIC` the function should be used when no interface is specified.

erebe replied:
Ok, will change

jacksgt commented on src/intern_blks/ya_intern.c:
```diff
@@ -277,14 +289,24 @@ void ya_int_bandwidth(ya_block_t * blk) {
+        if(rxfile) fclose(rxfile);
+        if(txfile) fclose(txfile);
+
+        fprintf(stdout, "Exiting bandwith thread\n");
```
Remove the debug output

erebe replied:
k

jacksgt commented on src/intern_blks/ya_intern.c:
```diff
@@ -207,62 +207,74 @@ void ya_int_brightness(ya_block_t *blk) {
+        char buffer[255] = { 0 };
+
+        route_h = fopen("/proc/net/route", "r");
+        if(route_h == NULL) goto fail;
```
Just do `if( ! route_h ) return;` (opening the file failed, therefore no need to close it)

erebe replied:
well catch

erebe commented:
Updated with your comments in mind.
I also removed the 2nd goto, as I added it since before my cleanup the function was more complex, now it is not needed anymore

jacksgt commented on src/intern_blks/ya_intern.c:
```diff
@@ -207,62 +207,72 @@ void ya_int_brightness(ya_block_t *blk) {
-		if(fscanf(txfile, "%lu", &tx)!=1)
-			fprintf(stderr, "Error getting values from file (%s)\n", txpath);
+	for(;;) {
+                if(strcmp(blk->internal->option[0], "") == 0) {
```
In case the config option is not set at all (NULL), we should also fallback to default:
```c
if (blk->internal->option[0] == NULL
    || strcmp(blk->internal->option[0], "") == 0) {
    ...
```

erebe commented:
Done

erebe commented:
Done

jacksgt commented:
Nice, thank you!
